metadata-ingestion/src/datahub/ingestion/source/unity/proxy_types.py

```diff
@@ -239,8 +249,8 @@ class Table(CommonProperty):
     properties: Dict[str, str]
     upstreams: Dict[TableReference, Dict[str, List[str]]] = field(default_factory=dict)
     external_upstreams: Set[ExternalTableReference] = field(default_factory=set)
-    upstream_notebooks: Set[NotebookId] = field(default_factory=set)
-    downstream_notebooks: Set[NotebookId] = field(default_factory=set)
+    upstream_notebooks: Set[NotebookReference] = field(default_factory=set)
+    downstream_notebooks: Set[NotebookReference] = field(default_factory=set)
```

Comment:
we use a set here to ensure that the refs are deduplicated. I don't think we'd want the last_updated timestamp to be considered when we're doing that deduplication